### PR TITLE
Use the cluster client object to create the apb pod

### PR DESCRIPTION
**Pull request by rthallisey**
Using oc run relies on the oc client. Instead, let's use
the cluster client to talk directly with kubernetes to spawn
the apb pod.

### CODE REVIEW

**Comment by eriknelson** [comment]
+1, this is inevitable and it's obvious the broker needs to move in this direction for better native integration. Thanks Ryan.

This is still good for a merge? I'd like to move towards using this API for creating ServiceAccounts and whatever other resources are required, this is a nice precedence.

**Comment by rthallisey** [comment]
@eriknelson Yup this is all set to merge.

**Comment by jmrodri** [comment]
@rthallisey please consult with @cfchase because he just added context support with namespace in it. See https://github.com/openshift/ansible-service-broker/pull/161

**Comment by cfchase** [comment]
@jmrodri Made requested changes.  Since you're unavailable, will pass review on to @eriknelson and @rthallisey.  @rthallisey I grabbed and added the namespace from the context being passed in from the catalog.  I think you could use these changes for your PR https://github.com/openshift/ansible-service-broker/pull/127.

File: pkg/apb/client.go [hunk 1]
```diff
@@ -174,20 +149,69 @@ func (c *Client) RunImage(
...
+		}
 	} else {
-		return RunCommand("oc", "run", fmt.Sprintf("aa-%s", uuid.New()),
-			"--env", fmt.Sprintf("OPENSHIFT_TARGET=%s", clusterConfig.Target),
-			"--env", fmt.Sprintf("OPENSHIFT_USER=%s", clusterConfig.User),
-			"--env", fmt.Sprintf("OPENSHIFT_PASS=%s", clusterConfig.Password),
-			fmt.Sprintf("--image-pull-policy=Always"),
-			fmt.Sprintf("--image=%s", spec.Image), "--restart=Never",
-			"--", action, "--extra-vars", string(params))
+		pod = &v1.Pod{
```
Comment (jwmatthews) [inline pkg/apb/client.go]: I'm concerned about the differentiation of running in cluster or out, feels wrong to need to make this choice in the code.

File: pkg/apb/client.go [hunk 2]
```diff
@@ -174,20 +149,69 @@ func (c *Client) RunImage(
...
+						}},
+						ImagePullPolicy: v1.PullAlways,
+					},
+				},
+				RestartPolicy: v1.RestartPolicyNever,
+			},
+		}
 	}
+
+	c.log.Notice(fmt.Sprintf("Creating pod %q in namespace default", pod.Name))
```
Comment (jwmatthews) [inline pkg/apb/client.go]: Is this correct that the pod is coming up in the 'namespace default'?

Comment (rthallisey) [inline pkg/apb/client.go]: Good catch. This should change based on the Namespace variable.

File: pkg/apb/client.go (hunk 1 above)
Comment (rthallisey) [inline pkg/apb/client.go]: 👍 Me too.

**Comment by eriknelson** [comment]
Ended up building #174 on top of this (thank you @rthallisey !!). We should probably close this one since its functionality is actually being brought in.

**Comment by rthallisey** [comment]
Closing in favor of #174

**Pull request closed without merging**